File: 파이썬/4.py
n=int(input())
arr=list(map(int,input().split()))
l=len(arr)
hap=sum(arr)
arr.sort()
for i in range(1,hap+1):
    h=i
    
    for j in range(l):
        if h-arr[l-1-j]>=0:
            h=h-arr[l-1-j]
    if h>0:
        print(i)
        

File: c++/4.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include<iostream>
#include<algorithm>
int* vis;
using namespace std;
int main(void) {
    
   
    int* arr;
    int n;
    int sum = 1;
    cin >> n;
    arr = new int[n];
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    sort(arr, arr + n);
    for (int i = 0; i < n; i++) {
        if (sum <arr[i]) {
            break;

        }
        sum += arr[i];
    }
    cout << sum;
    
    


}